docs:Updated bpf documentation files (#33964)
cilium-team/cilium: Documentation/bpf/toolchain.rst

    $ llc xdp-example.bc -march=bpf -mcpu=probe -filetype=obj -o xdp-example.o

Generally, LLVM IR generation is architecture independent. There are

> however, a few differences when using ``clang --target=bpf`` versus

Reviewer: “There are however,” sounds weird to me if you only have a comma _after_ “however”. Wouldn't you need one before as well? “There are, however,” or even “However/Nonetheless, there are a few differences...”.

          *c = sum;
        }

> At default code generation, the assembler will look like:

Reviewer: Since we're fixing this, let's just use present tense, “looks like”.

64-bit registers are used, hence the addition means 64-bit addition. Now, if you
enable the new 32-bit subregisters support by specifying ``-mattr=+alu32``, then

> the assembler will look like:

Reviewer: “looks like”

  It basically means that the ``bpf`` classifier does not need to call into external
  tc action modules, which are not necessary for ``bpf`` anyway, since all packet
  mangling, forwarding or other kind of actions can already be performed inside

>   the single BPF program, which is to be attached, and is therefore significantly

Reviewer: No, adding the comma here reads as if “the single BPF program” is “significantly faster”, when the real subject is “all packet mangling, forwarding or other kind of actions”.

Let's remove “which is to be attached” entirely, it doesn't bring any useful info here:

```suggestion
  a single BPF program, and is therefore significantly
```

    # tc qdisc del dev em1 clsact

  tc BPF programs can also be offloaded if the NIC and driver has support for it

>   similarly, as with XDP BPF programs. Netronome's nfp supported NICs offer both

Reviewer: Doesn't sound correct. I think this should be “... if the NIC and driver have support for it, like XDP BPF programs.”
